src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -752,7 +751,7 @@ public static int onBlockBreakEvent(World world, GameType gameType, EntityPlayer
     {
         // Logic from tryHarvestBlock for pre-canceling the event
         boolean preCancelEvent = false;
-        if (gameType.isCreative() && !entityPlayer.getHeldItemMainhand().isEmpty() && entityPlayer.getHeldItemMainhand().getItem() instanceof ItemSword)
+        if (gameType.isCreative() && !entityPlayer.getHeldItemMainhand().isEmpty() && !entityPlayer.getHeldItemMainhand().getItem().canDestroyBlockInCreative(entityPlayer.getHeldItemMainhand(), world.getBlockState(pos), entityPlayer))
```

Comment:
I think it should be tidied up, I suggested the change not for performance, but because the line is really excessively long now. It's better if you can actually see what this line is doing on the same page as the rest of it.